src/geometry.rs

```diff
@@ -38,27 +64,47 @@ impl<T> Rect<T>
 where
     T: Add<Output = T> + Copy + Clone,
 {
-    pub(crate) fn horizontal(&self) -> T {
+    /// The sum of [`Rect.left`](Rect) and [`Rect.right](Rect)
```

Comment:
Typo:
```suggestion
    /// The sum of [`Rect.left`](Rect) and [`Rect.right`](Rect)
```
However, I think this is misleading, because a `Rect` doesn't have `left` or `right` attributes.
```suggestion
    /// The sum of [`Rect.start`](Rect) and [`Rect.end`](Rect)
```